hazelcast/src/main/java/com/hazelcast/internal/metrics/jmx/JmxPublisher.java

```diff
@@ -171,25 +189,29 @@ public void whenComplete() {
     // package-visible for test
     @SuppressWarnings("checkstyle:BooleanExpressionComplexity")
     static String escapeObjectNameValue(String name) {
```

Comment:
What about this? 
```java
    @SuppressWarnings({"checkstyle:BooleanExpressionComplexity", "checkstyle:CyclomaticComplexity"})
    static String escapeObjectNameValue(String name) {
        int l = name.length();
        int i;
        for (i = 0; i < l; i++) {
            char ch = name.charAt(i);
            if (ch == '=' || ch == ':' || ch == '*' || ch == '?' || ch == '"' || ch == '\n') {
                break;
            }
        }
        if (i == l) {
            return name;
        }
        StringBuilder res = new StringBuilder(name.length() + (1 << 3));
        res.append('"');
        for (i = 0; i < l; i++) {
            char ch = name.charAt(i);
            if (ch == '\\' || ch == '"' || ch == '*' || ch == '?') {
                res.append('\\');
            }
            if (ch == '\n') {
                res.append("\\n");
            } else {
                res.append(ch);
            }
        }
        res.append('"');
        return res.toString();
    }
```

Reply:
@viliam-durina I belive that your approach is better. I just splitted it into two methods.